### ISSUE TITLE
es-internal-lb was not found

### ISSUE THREAD

**Issue by MedAnd** [elastic/azure-marketplace#154]
Deploying to a new resource group using an existing Vnet fails

es-internal-lb was not found. Please make sure that the referenced resource exists, and that both resources are in the same region

**Comment by russcam** [comment]
Hey @MedAnd 

1. How are you deploying? Azure Marketplace? From this repository (if so, which release tag)?
2. Is the new resource group in the same region as the VNet?
3. Is the es-internal-lb in the same region as the VNet?
4. Does the subnet on the VNet to which the deployed resources will be attached have sufficient addresses for all resources?
5. Can you provide the deployment log?

If you can provide a small succinct example that reproduces the issue, it will help to diagnose what the issue is much quicker.

Thanks!

**Comment by MedAnd** [comment]
Hi @russcam,

1. So I was deploying using the deploy button at the bottom of the page, which is a custom template from Elastic but still within the Azure portal experience. I assume the button uses the latest github release?
2. Yes
3. The es-internal-lb did not exist but I wanted it to be in the same region. Assume the json template creates it.
4. The subnet did not exist, I assumed the template would create it. Once I created the subnet though the deployment is now progressing. 

So in summary the it seems the subnet must exist while the load balancer will be created, the template will not create it... wonder if the docs or template could be made a little more explicit, stating which resources get auto created & which must exist?

**Comment by russcam** [comment]
Yes, subnets for the cluster and for Application Gateway (if choosing `gateway` for load balancing) must exist already when deploying into an `existing` VNet. The template does not make any changes to an existing VNet.

For the resources in a resource group, they can be deployed to the same region as the resource group which is the default, or to another region where those resources are available for deployment.

I've updated the README to be more explicit 👍
